### ISSUE TITLE
Takeoff with mavlink command causes the drone scratching with the ground

### ISSUE THREAD

**Issue by Lqs66** [PX4/PX4-Autopilot#21622]
## Describe the bug
1. When the drone is in takeoff mode, using Mavlink to send the MAV_CMD_NAV_TAKEOFF command may result in one of the following abnormal behaviors:
1a. The blades are spinning, but the speed is too slow, making it impossible for the drone to get off the ground;
1b. The drone can leave the ground, but the altitude is too low, and it will scratch the ground;
1c. Other abnormal flight behaviors (see flight logs below);

2. When the drone accepts the MAV_CMD_NAV_TAKEOFF command for the first time before takeoff, it may produce one of the following behaviors depending on the distance between the command setpoint and the home coordinates:
2a. If the setpoint is set to the home coordinate, the drone takes off as expected;
2b. If there is a horizontal distance between two coordinates (e.g., 11 m), the drone will move horizontally before reaching a safe altitude, which leads to a scratch between the airframe and the ground; (I suggest that the drone should fly to a safe altitude before moving horizontally.)
2c. If the distance becomes further, the drone can't leave the ground ( same behavior as 1a).

## To Reproduce

For bug 1:
1. make px4_sitl gazebo
2. Enter takeoff mode in the QGroundControl
3. Before the drone leaves the ground, execute the following code to send the MAV_CMD_NAV_TAKEOFF command

For bug 2:
1. modify the latitude and longitude in the code
a) To reproduce 2b, set latitude and longitude to (47.397851, 8.545607), respectively;
b) To reproduce 2c, set to (48, 8).
2. make px4_sitl gazebo
3. arm the drone in the QGroundControl
4. execute the code to send the MAV_CMD_NAV_TAKEOFF command

```
#!/usr/bin/env python3

import os, sys
import time
from pymavlink import mavutil

os.environ["MAVLINK20"] = "1"

if __name__ == "__main__":
    
    uav = mavutil.mavlink_connection("udpin:127.0.0.1:14550")
    uav.wait_heartbeat()

    print(f"send MAV_CMD_NAV_TAKEOFF")
    uav.mav.command_long_send(
        uav.target_system,
        uav.target_component,
        mavutil.mavlink.MAV_CMD_NAV_TAKEOFF,
        0,
        0,
        0,
        0,
        0,
        48,  # latitude
        8,   # longitude
        100  # ASL altitude
    )
```

## Expected behavior
Both takeoff procedures should have the same behavior described in the takeoff mode documentation (see https://docs.px4.io/v1.13/en/flight_modes/takeoff.html).

## Log Files and Screenshots

1a (Failed to takeoff): https://logs.px4.io/plot_app?log=50e0e9fa-96ba-478b-8524-533b2f22d365
1b (Scratching with the ground): https://logs.px4.io/plot_app?log=791a9860-1eca-48f2-8a3e-575de70051e9
1c (Other): https://logs.px4.io/plot_app?log=e6ed2e26-d240-4b1b-aa6f-6c860ac4fe1f
2b (Horizontal movement before reaching the safe altitude): https://logs.px4.io/plot_app?log=88042ddf-17b4-49e1-a43d-964f34b4a30f
2c (Failed to takeoff): https://logs.px4.io/plot_app?log=2512581a-325d-4d81-888b-6c9e28d5a141

## Drone:
- Default PX4 quadcopter in Gazebo
- PX4 autopilot version: v1.13.2 and v1.13.3

**Comment by antonio-sc66** [comment]
I might be wrong, but I think that the takeoff latitude and longitude have to be the current ones or really close, I don't think it expects you to send a distant point. If you want to takeoff and then move to a different point you should perform a takeoff and a goto afterwards or something similar, at least that's how I do it.

**Comment by Lqs66** [comment]
> Hi @Lqs66, I might be wrong, but I think that the takeoff latitude and longitude have to be the current ones or really close, I don't think it expects you to send a distant point. If you want to takeoff and then move to a different point you should perform a takeoff and a goto afterwards or something similar, at least that's how I do it.
> 
> QGroundControl doesn't even let you choose a takeoff point, it reads it from the current vehicle positions and fills that value

You may be right. But the navigator module will allow coordinates from the takeoff command.
What do the developers have to say about this?

```
if (PX4_ISFINITE(cmd.param5) && PX4_ISFINITE(cmd.param6)) {
	rep->current.lat = cmd.param5;
	rep->current.lon = cmd.param6;

} else {
	// If one of them is non-finite set the current global position as target
	rep->current.lat = get_global_position()->lat;
	rep->current.lon = get_global_position()->lon;
}
```

**Comment by sfuhrer** [comment]
>You may be right. But the navigator module will allow coordinates from the takeoff command.
What do the developers have to say about this?

For me it's also unclear why the Takeoff should have a lat/lon attribute. But the mavlink command has those fields so guess we have to support it? @MaEtUgR what's your take on it?

About your issues @Lqs66:
- did you set the takeoff on purpose to 100m AMSL? The drone in simulation is spawned at 490m AMSL, so well above that.
- still, it should takeoff to the minimum takeoff altitude (2.5m), but doesn't seem to do so: 
![image](https://github.com/PX4/PX4-Autopilot/assets/26798987/a97340c2-181d-4dc0-9c0c-a5db29c77797)
- the "move laterally and doesn't climb vertically" sounds like something that maybe is fixed with https://github.com/PX4/PX4-Autopilot/pull/20908?

**Comment by Lqs66** [comment]
> > You may be right. But the navigator module will allow coordinates from the takeoff command.
> > What do the developers have to say about this?
> 
> For me it's also unclear why the Takeoff should have a lat/lon attribute. But the mavlink command has those fields so guess we have to support it? @MaEtUgR what's your take on it?
> 
> About your issues @Lqs66:
> 
> * did you set the takeoff on purpose to 100m AMSL? The drone in simulation is spawned at 490m AMSL, so well above that.
> * still, it should takeoff to the minimum takeoff altitude (2.5m), but doesn't seem to do so. The local position setpoint in z stays very low:
>   ![image](https://user-images.githubusercontent.com/26798987/253932699-a97340c2-181d-4dc0-9c0c-a5db29c77797.png)
> * the "move laterally and doesn't climb vertically" sounds like something that maybe is fixed with [Navigator: remove _can_loiter_at_sp and _need_takeoff #20908](https://github.com/PX4/PX4-Autopilot/pull/20908)?

Yes, I set the takeoff on purpose to 100m , but this altitude should be the height from the horizontal. It seems that in PX4 the minimum takeoff height (2.5m) is the height from the ground, which is different from the altitude. That means 2.5m will first be converted to altitude (roughly 490m) and then compared to 100m, which is obviously less than 490m, so drone will use 490m for takeoff altitude. Not sure if I'm understanding this correctly?
